app/src/test/java/com/swent/suddenbump/model/user/UserRepositoryFirestoreTest.kt

```diff
@@ -2922,4 +2922,507 @@ class UserRepositoryFirestoreTest {
+    val sharedWithUidList = listOf(friend1.uid, friend2.uid)
+
+    // Mock document snapshots for each friend
+    val friend1Snapshot = mock(DocumentSnapshot::class.java)
+    val friend2Snapshot = mock(DocumentSnapshot::class.java)
+
+    `when`(mockUserDocumentSnapshot.data)
+        .thenReturn(mapOf("locationSharedWith" to sharedWithUidList))
+
+    // Mock document references for each friend
+    val friend1DocumentRef = mock(DocumentReference::class.java)
+    val friend2DocumentRef = mock(DocumentReference::class.java)
+
+    `when`(mockFirestore.collection("Users").document(friend1.uid)).thenReturn(friend1DocumentRef)
+    `when`(friend1DocumentRef.get()).thenReturn(Tasks.forResult(friend1Snapshot))
+
+    `when`(mockFirestore.collection("Users").document(friend2.uid)).thenReturn(friend2DocumentRef)
+    `when`(friend2DocumentRef.get()).thenReturn(Tasks.forResult(friend2Snapshot))
+
+    `when`(friend1Snapshot.exists()).thenReturn(true)
+    `when`(friend2Snapshot.exists()).thenReturn(true)
+
+    // Mock documentSnapshotToUser behavior
+    val helper = mock(UserRepositoryFirestoreHelper::class.java)
+    `when`(helper.documentSnapshotToUser(friend1Snapshot, null)).thenReturn(friend1)
+    `when`(helper.documentSnapshotToUser(friend2Snapshot, null)).thenReturn(friend2)
+
+    // Inject the mock helper into the repository
+    val helperField = UserRepositoryFirestore::class.java.getDeclaredField("helper")
+    helperField.isAccessible = true
+    helperField.set(userRepositoryFirestore, helper)
+
+    // Prepare tasks for whenAllSuccess
+    val friendTasks = listOf(friend1DocumentRef.get(), friend2DocumentRef.get())
+
+    // Mock Tasks.whenAllSuccess behavior
+    `when`(Tasks.whenAllSuccess<DocumentSnapshot>(friendTasks))
+        .thenReturn(Tasks.forResult(listOf(friend1Snapshot, friend2Snapshot)))
+
+    // Invoke the method under test
+    userRepositoryFirestore.getSharedWithFriends(
+        uid = "user123",
+        onSuccess = { result ->
+          // Verify the result
+          assert(result.size == 2)
+          assert(result.contains(friend1))
+          assert(result.contains(friend2))
+        },
+        onFailure = { assert(false) { "Should not fail on success" } })
+  }
+
+  @Test
+  fun getSharedWithFriendsReturnsEmptyListWhenNoData() {
+    val uid = "user123"
+
+    // Mock document snapshot for the user
+    val mockUserSnapshot = mock(DocumentSnapshot::class.java)
+    `when`(mockUserDocumentReference.get()).thenReturn(Tasks.forResult(mockUserSnapshot))
+
+    // Mock the data to simulate no "locationSharedWith" field
+    `when`(mockUserSnapshot.data).thenReturn(emptyMap())
+
+    userRepositoryFirestore.getSharedWithFriends(
+        uid = uid,
+        onSuccess = { result ->
+          assertNotNull(result)
+          assertTrue(result.isEmpty())
+        },
+        onFailure = { fail("Should not fail when no data is available") })
+
+    shadowOf(Looper.getMainLooper()).idle()
+  }
```

Comment:
Tests are very extensive, and also the comments makes them easy to read. Nice job!